src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackFrame.cs

```diff
@@ -57,6 +58,7 @@ private void InitMembers()
         /// <summary>
         /// Constructs a StackFrame corresponding to the active stack frame.
         /// </summary>
+        [MethodImplAttribute(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
```

Comment:
Do we need `NoOptimization` here? Other places in this PR (and also existing like `Environment.StackTrace`) just use `NoInlining`.

Reply:
Without it the  `call InitializeForCurrentThread` can/gets be replaced with a ` jmp`.

Reply:
Hmmm, we are missing the NoInlining check in `CorInfoImpl.*.cs` `canTailCall`.